docs/AA--Javascript-in-EE/value-objects/label.md

```diff
@@ -0,0 +1,168 @@
+
+#### `Label.fromSameSingleAndPlural`
+This is used to construct an instance of Label using the passed in argument for both singular and plural forms of the label.  This is useful when a label has no distinction between its singular and plural usage.
+
+**Example:**
+
+```js
+const approvedLabel = Label.fromSameSingleAndPlural( 'approved' );
+
+// displays 'approved' in both cases
+console.log( approvedLabel.singular );
+console.log( approvedLabel.plural );
+```
+
+### _Properties_
+#### `Label.FORMAT_LOWERCASE`
+
+This static property on `Label` can be used to reference the lowercase format string.
+
+#### `Label.FORMAT_UPPERCASE`
+
+This static property on `Label` can be used to reference the uppercase format string.
+
+#### `Label.FORMAT_SENTENCE_CASE`
+
+This static property on `Label` can be used to reference the sentence case format string.
+
+### Instance
+
+The following methods and properties exist after instantiating a `Label`.
+
+### _Methods_
+
+#### `setSingular( singular )`
+
+This is a fluid setter for setting the singular property on the `Label`. If the singular property has already been set, this will return a new instance of Label (and the plural property will be an equal value as the current instance).
+
+**Example**:
+
+```js
+const myLabel = new Label( 'bird', 'birds' );
+const myNewLabel = myLabel.setSingular( 'birdy' );
+
+// returns 'bird'
+console.log( myLabel.singular );
+
+// returns 'birdy'
+console.log( myNewLabel.singular );
+
+// returns 'birds'
+console.log( myNewLabel.plural );
+```
+
+#### `setPlural( plural )`
+
+This is a fluid setter for setting the plural property on the `Label`.  If the plural property has already been set, this will return a new instance of `Label` (and the singular property will be an equal value as the current instance).
+
+**Example**:
+
+```js
+const myLabel = new Label( 'bird', 'birds' );
+const myNewLabel = myLabel.setPlural( 'flock' );
+
+// returns 'birds'
+console.log( myLabel.plural );
+
+// returns 'bird'
+console.log( myNewLabel.singular );
+
+// returns 'flock'
+console.log( myNewLabel.plural );
+```
+
+#### `asSentenceCase( singular = true )`
+
+Returns the value for the indicated property formatted in sentence case.  By default returns the value for the singular property, however if you pass in anything but `true` as the argument, then the plural property value is used and returns as sentence case.
+
+> Note, this strips any `-` in dashed labels.  So for instance if your label value was `something-else`, the value returned would be `Something Else`.
+
+**Example**:
+
+```js
+const myLabel = new Label( 'pack of gum', 'packs of gum' );
+
+// 'Pack Of Gum'
+console.log( myLabel.asSentenceCase() );
+
+// 'Packs of Gum'
+console.log( myLabel.asSentenceCase( false ) );
+```
+
+#### `asLowerCase( singular = true )`
+
+Returns the value for the indicated property formatted in lower case.
```

Comment:
in some places you use "lower case" and in others (like the Properties section) you use "lowercase",
either is correct, but we should stick with one or the other.